pkg/kots/rego/kots-spec-opa-nonrendered.rego

```diff
@@ -553,6 +566,119 @@ lint[output] {
   }
 }
 
+# Check if repeatable ConfigOption has a template field defined
+lint[output] {
+  config_option := config_options[_]
+  item := config_option.item
+  item.repeatable
+  not item.template
+  field := concat(".", [config_option.field, "type"])
+  message := sprintf("Repeatable Config option \"%s\" is missing the template field", [string(item.name)])
+  output := {
+    "rule": "repeat-config-option-missing-template",
+    "type": "error",
+    "message": message,
+    "path": config_file_path,
+    "field": field,
+    "docIndex": config_data.docIndex
+  }
+}
+
+# Check if repeatable ConfigOption has at least one template
+lint[output] {
+  config_option := config_options[_]
+  item := config_option.item
+  item.repeatable
+  item.template == null
+  field := concat(".", [config_option.field, "type"])
+  message := sprintf("Repeatable Config option \"%s\" is missing a template target", [string(item.name)])
+  output := {
+    "rule": "repeat-config-option-missing-template",
+    "type": "error",
+    "message": message,
+    "path": config_file_path,
+    "field": field,
+    "docIndex": config_data.docIndex
+  }
+}
+
+# Check if repeatable ConfigOption has a valuesByGroup field
+lint[output] {
+  config_option := config_options[_]
+  item := config_option.item
+  item.repeatable
+  not item.valuesByGroup
+  field := concat(".", [config_option.field, "type"])
+  message := sprintf("Repeatable Config option \"%s\" is missing the valuesByGroup field", [string(item.name)])
+  output := {
+    "rule": "repeat-config-option-missing-valuesByGroup",
+    "type": "error",
+    "message": message,
+    "path": config_file_path,
+    "field": field,
+    "docIndex": config_data.docIndex
+  }
+}
+
+# Check if repeatable ConfigOption has at least one group in valuesByGroup
+lint[output] {
+  config_option := config_options[_]
+  item := config_option.item
+  item.repeatable
+  item.valuesByGroup
+  item.valuesByGroup == null
```

Comment:
this negates the previous line, i don't get how this is working, if `item.valuesByGroup` is defined, how is it ever gonna be `null`?

Reply:
i think you should be checking for the object length here instead?

Reply:
item.valuesByGroup checks that item has a field named "valuesByGroup".  However, if valuesByGroup contains no entries, rego considers it "null" because it doesn't know what type it is and there are no values provided.

If the item entry does not have a 'valuesByGroup' field at all (it's set to omitempty in the struct definition), it will stop processing on line 628.  This is shown in the tests as such:
``` yaml
- name: valuesByGroup_with_no_group
  type: text
  repeatable: true
  template:
  - name: example-config
  valuesByGroup:
```

Reply:
I was originally checking with count, but if there are 0 entries rego doesn't understand that it's even an array.  You'll get this error:
```
FAIL: Test_lintWithOPANonRendered (0.12s)
    --- FAIL: Test_lintWithOPANonRendered/repeatable_config_spec (0.03s)
        lint_test.go:1439: 
                Error Trace:    lint_test.go:1439
                Error:          Received unexpected error:
                                kots-spec-opa-nonrendered.rego:592: eval_type_error: count: operand 1 must be one of {array, object, set} but got null
                                failed to evaluate query
```

using this as the test:
```
# Check if repeatable ConfigOption has at least one template
lint[output] {
  config_option := config_options[_]
  item := config_option.item
  item.repeatable
  count(item.template) == 0
  field := concat(".", [config_option.field, "type"])
  message := sprintf("Repeatable Config option \"%s\" is missing a template target", [string(item.name)])
  output := {
    "rule": "repeat-config-option-missing-template-entry",
    "type": "error",
    "message": message,
    "path": config_file_path,
    "field": field,
    "docIndex": config_data.docIndex
  }
}
```

Reply:
removed after updating schema